iothub/device/src/ClientPropertyCollection.cs

```diff
@@ -180,35 +181,72 @@ public virtual bool TryGetValue<T>(string componentName, string propertyName, ou
                     $"TryGetValue will attempt to get the property value but may not behave as expected.", nameof(TryGetValue));
             }
 
+            // If either the component name or the property name is null, empty or whitespace,
```

Comment:
This is self explanatory in the code. Might want to say  - If the component name is not provided, return false with default value of type <T> passed in.